src/manifest.rs

```diff
@@ -183,6 +164,79 @@ impl Manifest {
         data.parse()
     }
 
+    /// Get the specified table from the manifest.
+    pub fn get_table<'a>(
+        &'a mut self,
+        table_path: &[String],
+    ) -> Result<&'a mut BTreeMap<String, toml::Value>, ManifestError> {
+
+        /// Descend into a manifest until the required table is found.
+        fn descend<'a>(
+            input: &'a mut BTreeMap<String, toml::Value>,
+            mut path: VecDeque<&String>,
+        ) -> Result<&'a mut BTreeMap<String, toml::Value>, ManifestError> {
+            if let Some(segment) = path.pop_front() {
+                let value = match *input
+                    .entry(segment.to_owned())
+                    .or_insert_with(|| toml::Value::Table(BTreeMap::new())) {
+                    toml::Value::Table(ref mut t) => t,
+                    _ => return Err(ManifestError::NonExistentTable(segment.clone())),
+                };
+
+                descend(value, path)
+            } else {
+                Ok(input)
+            }
+        }
+
+        descend(&mut self.data, table_path.into_iter().collect())
+    }
+
+    /// Get all sections in the manifest that exist and might contain dependencies.
+    pub fn get_sections(&self) -> Vec<(Vec<String>, BTreeMap<String, toml::Value>)> {
+        // Dependencies can be in the three standard sections...
+        let mut sections = ["dev-dependencies", "build-dependencies", "dependencies"]
+            .into_iter()
+            .filter_map(|section_candidate| {
+                let section_candidate = section_candidate.to_string();
+
+                match self.data.get(&section_candidate) {
+                    Some(&toml::Value::Table(ref table)) => {
+                        Some((vec![section_candidate], table.clone()))
+                    }
+                    _ => None,
+                }
+
+            })
+            .collect::<Vec<_>>();
+
+        // ... or in `target.<target name>.(dev-)dependencies`.
```

Comment:
[Looks like](https://github.com/rust-lang/cargo/blob/d2d6e675fcb70cfb91c2cf2fad8cdd345c120316/tests/build.rs#L1778) target build-dependencies is also a thing.